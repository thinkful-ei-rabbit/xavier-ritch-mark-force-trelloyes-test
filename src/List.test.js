import React from "react";
import ReactDOM from "react-dom";
import List from "./List";
import renderer from 'react-test-renderer';

describe("<List />", () => {
    it("renders List without crashing", () => {
      const div = document.createElement("div");
  
      ReactDOM.render(<List
                      header="header text"
                      cards={[
                        {id: 'a', title: 'test title', content: 'test content',},
                        {id: 'b', title: 'test title', content: 'test content',}
                      ]}
                      />, div);
  
      // clean up code
      ReactDOM.unmountComponentAtNode(div);
    });
    it("renders UI", () => {
      const tree = renderer
          .create(<List 
            header="header text"
            cards={[
              {id: 'a', title: 'test title', content: 'test content',},
              {id: 'b', title: 'test title', content: 'test content',}
            ]}/>).toJSON();
          expect(tree).toMatchSnapshot();
    });
  });