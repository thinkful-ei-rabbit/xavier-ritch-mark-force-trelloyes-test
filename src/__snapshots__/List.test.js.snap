// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders UI 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header text
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test title
      </h3>
      <p>
        test content
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test title
      </h3>
      <p>
        test content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
